### PR TITLE
Add packager for FreeBSD (pkgng)

### PR DESCRIPTION
**Pull request by amdprophet**
### Description

Adds a new packager for FreeBSD (pkgng).

---

/cc @chef/omnibus-maintainers <- This ensures the Omnibus Maintainers team are notified to review this PR.
#### Maintainers

Please ensure that you check for:
- [] If this change impacts git cache validity, it bumps the git cache
  serial number
- [] If this change impacts compatibility with omnibus-software, the
  corresponding change is reviewed and there is a release plan
- [] If this change impacts compatibility with the omnibus cookbook, the
  corresponding change is reviewed and there is a release plan

### CODE REVIEW

File: lib/omnibus/packagers/pkgng.rb [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+    #     }
+    #   }
+    #
+    # @param [Hash] val
+    #   the dependencies for this package
+    #
+    # @return [Hash]
+    #   the dependencies for this package
+    #
+    def dependencies(val = NULL)
```
Comment (yzl) [inline lib/omnibus/packagers/pkgng.rb]: If this is for runtime dependencies, I'd rather call it `runtime_dependencies`; the `omnibus` DSL uses `dependency` and `dependencies` for buildtime dependencies.

Comment (amdprophet) [inline lib/omnibus/packagers/pkgng.rb]: I've renamed this to `runtime_dependencies` in the latest commit.

**Comment by schisamo** [comment]
@amdprophet Please squash everything down into a single commit and ensure it is signed off per the [Chef DCO](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

**Comment by amdprophet** [comment]
@schisamo squashed and signed off, thanks!

**Comment by schisamo** [comment]
@chef/client-freebsd I would like y'all to give this a review also.

File: lib/omnibus/packager.rb [hunk 2]
```diff
@@ -48,6 +49,7 @@ module Packager
       "ips"      => IPS,
       "windows"  => [MSI, APPX],
       "mac_os_x" => PKG,
+      "freebsd"  => PKGNG,
```
Comment (schisamo) [inline lib/omnibus/packager.rb]: To ensure some backward compat let's produce both a `Makeself` and `PKGNG` artifact on FreeBSD. The following code should work (be sure to update the specs also):

```ruby
"freebsd"  => [Makeself, PKGNG],
```

File: lib/omnibus/packagers/pkgng.rb [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+      File.open(destination, "w+") do |f|
+        FFI_Yajl::Encoder.encode(compact_manifest.merge(manifest), pretty: true).each_line do |line|
+          f.write(line)
+        end
+      end
+    end
+
+    #
+    # Create the +.txz+ file, compressing at xz level 9. The use of the
+    # +fakeroot+ command is required so that the package is owned by
```
Comment (srenatus) [inline lib/omnibus/packagers/pkgng.rb]: Is this comment still accurate? I don't see any `fakeroot` invocation there... But I could be missing something.

File: lib/omnibus/packagers/pkgng.rb [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+      destination = File.join(staging_dir, "+MANIFEST")
+      File.open(destination, "w+") do |f|
+        FFI_Yajl::Encoder.encode(compact_manifest.merge(manifest), pretty: true).each_line do |line|
+          f.write(line)
+        end
+      end
+    end
+
+    #
+    # Create the +.txz+ file, compressing at xz level 9. The use of the
```
Comment (srenatus) [inline lib/omnibus/packagers/pkgng.rb]: Would we want to make this configurable? If FreeBSD is commonly using xz level 9, I suppose we could, too. However, the xz command line utilities warn about level 7-9:

```
  -0 ... -9           compression preset; default is 6; take compressor *and*
                      decompressor memory usage into account before using 7-9!
```

I couldn't find quickly what `pkg create` would end up using...

File: lib/omnibus/packagers/pkgng.rb (hunk 3 above)
Comment (scotthain) [inline lib/omnibus/packagers/pkgng.rb]: Guessing it's a copypasta from https://github.com/chef/omnibus/blob/master/lib/omnibus/packagers/deb.rb#L363-L365 ?

Comment (amdprophet) [inline lib/omnibus/packagers/pkgng.rb]: It was indeed copypasta. I'm not sure on a default for `pkg create` off the top of my head. I'll look into it.

File: lib/omnibus/packagers/pkgng.rb (hunk 4 above)
Comment (amdprophet) [inline lib/omnibus/packagers/pkgng.rb]: I mistakingly commented that `pkg create` was using a default xz compression level of 9. The man pages for `xz` shows that the default is 6 which should be what's being used here.

**Comment by scotthain** [comment]
Looks like we need to add a `pkg_safe` sort of method (similar to the `rpm_safe` method here: https://github.com/chef/omnibus/blob/master/lib/omnibus/packagers/rpm.rb#L526-L534) to ensure that if any files have restricted characters, they are substituted. I'm also running into a strange issue with symlinks not picking up correctly, but I think that may be my test setup.

**Comment by amdprophet** [comment]
I'm going to close this as we aren't using this project anymore.

**Pull request closed without merging**